packages/react/src/Banner/Banner.tsx

```diff
@@ -0,0 +1,323 @@
+      if (!title) {
+        throw new Error(
+          'The Banner component requires a title to be provided as the `title` prop or through `Banner.Title`',
+        )
+      }
+    }, [titleId])
+  }
+
+  return (
+    <BannerContext.Provider value={value}>
+      <StyledBanner {...rest} aria-labelledby={titleId} as="section" data-variant={variant} tabIndex={-1} ref={ref}>
+        <div className="BannerIcon">{icon && variant === 'info' ? icon : iconForVariant[variant]}</div>
+        <div className="BannerContainer">
+          <div className="BannerContent">
+            {title ? <BannerTitle>{title}</BannerTitle> : null}
+            {description ? <BannerDescription>{description}</BannerDescription> : null}
+            {children}
+          </div>
+          {hasActions ? <BannerActions primaryAction={primaryAction} secondaryAction={secondaryAction} /> : null}
+        </div>
+        {dismissible ? (
+          <IconButton
+            aria-label="Dismiss banner"
+            onClick={onDismiss}
+            className="BannerDismiss"
+            icon={XIcon}
+            variant="invisible"
+          />
+        ) : null}
+      </StyledBanner>
+    </BannerContext.Provider>
+  )
+})
+
+/**
+ * For styling, it's important that the icons and the text have the same height
+ * for alignment to occur in multi-line scenarios. Currently, we use a
+ * line-height of `20px` so that means that the height of icons should match
+ * that value.
+ */
+const StyledBanner = styled.div`
+  display: grid;
+  grid-template-columns: auto minmax(0, 1fr) auto;
+  align-items: start;
+  background-color: var(--banner-bgColor);
+  border: 1px solid var(--banner-borderColor);
+  padding: 0.5rem;
+  border-radius: ${get('radii.2')};
+
+  &[data-variant='critical'] {
+    --banner-bgColor: ${get('colors.danger.subtle')};
+    --banner-borderColor: ${get('colors.danger.muted')};
+    --banner-icon-bgColor: ${get('colors.danger.fg')};
+  }
+
+  &[data-variant='info'] {
+    --banner-bgColor: ${get('colors.accent.subtle')};
+    --banner-borderColor: ${get('colors.accent.muted')};
+    --banner-icon-bgColor: ${get('colors.accent.fg')};
+  }
+
+  &[data-variant='success'] {
+    --banner-bgColor: ${get('colors.success.subtle')};
+    --banner-borderColor: ${get('colors.success.muted')};
+    --banner-icon-bgColor: ${get('colors.success.fg')};
+  }
+
+  &[data-variant='upsell'] {
+    --banner-bgColor: ${get('colors.done.subtle')};
+    --banner-borderColor: ${get('colors.done.muted')};
+    --banner-icon-bgColor: ${get('colors.done.fg')};
+  }
+
+  &[data-variant='warning'] {
+    --banner-bgColor: ${get('colors.attention.subtle')};
+    --banner-borderColor: ${get('colors.attention.muted')};
+    --banner-icon-bgColor: ${get('colors.attention.fg')};
+  }
+
+  /* BannerIcon ------------------------------------------------------------- */
+
+  .BannerIcon {
+    display: grid;
+    place-items: center;
+    padding: 0.5rem;
```

Comment:
```suggestion
    padding: 0.375rem 0.5rem;
```
I'm pulling this from PVC, where we went pretty specific here to align the title with the trailing actions labels. Do you hate it? Is it okay? I think it's acceptable 😆